tools/integration_tests/implicit_dir/local_file_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package implicit_dir
+
+import (
+	"io/fs"
+	"path"
+	"path/filepath"
+	"testing"
+
+	. "github.com/googlecloudplatform/gcsfuse/tools/integration_tests/local_file/helpers"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+func TestNewFileUnderImplicitDirectoryShouldNotGetSyncedToGCSTillClose(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+	// Create file in implicit directory.
+	err := CreateObject(setup.TestBucket(), path.Join(ImplicitDirName, ImplicitFileName1), FileContents)
+	if err != nil {
+		t.Errorf("Error while creating implicit directory, err: %v", err)
+	}
+
+	// Validate.
+	NewFileShouldGetSyncedToGCSAtClose(path.Join(ImplicitDirName, FileName1), t)
+}
+
+func TestReadDirForImplicitDirWithLocalFile(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+	// Create implicit dir with 2 local files and 1 synced file.
+	err := CreateObject(setup.TestBucket(), path.Join(ImplicitDirName, ImplicitFileName1), "")
+	if err != nil {
+		t.Errorf("Error while creating implicit directory, err: %v", err)
+	}
+	_, fh1 := CreateLocalFile(path.Join(ImplicitDirName, FileName1), t)
+	_, fh2 := CreateLocalFile(path.Join(ImplicitDirName, FileName2), t)
+
+	// Attempt to list implicit directory.
+	entries := ReadDirectory(path.Join(setup.MntDir(), ImplicitDirName), t)
+
+	// Verify entries received successfully.
+	VerifyCountOfEntries(3, len(entries), t)
+	VerifyLocalFileEntry(entries[0], FileName1, 0, t)
+	VerifyLocalFileEntry(entries[1], FileName2, 0, t)
+	VerifyLocalFileEntry(entries[2], ImplicitFileName1, 0, t)
+	// Close the local files.
+	CloseFileAndValidateObjectContents(fh1, path.Join(ImplicitDirName, FileName1), "", t)
+	CloseFileAndValidateObjectContents(fh2, path.Join(ImplicitDirName, FileName2), "", t)
+}
+
+func TestRecursiveListingWithLocalFiles(t *testing.T) {
+	// Structure
```

Comment:
Please write structure like [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/operations/copy_dir_test.go#L126). This is very confusing.

Reply:
Please fix file,directory in one line if possible.

Reply:
It is a github issue. On intelliJ it shows in one line only.